ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -203,23 +218,82 @@ func (h *FaultHandler) CheckNetworkBlackHolePort() func(http.ResponseWriter, *ht
 		rwMu.RLock()
 		defer rwMu.RUnlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Return the correct status state
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("running")
-		logger.Info("Successfully checked status for fault", logger.Fields{
-			field.RequestType: requestType,
-			field.Request:     request.ToString(),
-			field.Response:    responseBody.ToString(),
-		})
+		ctx := context.Background()
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, requestTimeoutDuration)
+		defer cancel()
+
+		var responseBody types.NetworkFaultInjectionResponse
+		var statusCode int
+		chainName := fmt.Sprintf("%s-%s-%s", *request.TrafficType, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+		running, cmdOutput, cmdErr := h.checkNetworkBlackHolePort(ctxWithTimeout, request, chainName, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+
+		// We've timed out trying to check if the black hole port fault injection is running
+		if err := ctx.Err(); err != nil {
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(requestTimedOutError)
+		} else if cmdErr != nil {
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(cmdOutput)
+		} else {
+			statusCode = http.StatusOK
+			if running {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("running")
+			} else {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("not-running")
+			}
+			logger.Info("Successfully checked status for fault", logger.Fields{
```

Comment:
nit: can we move this log out of the block so that it can cover the other two use cases above as well?

Reply:
Was thinking that we should not output this log statement if we either timeout or run into some internal error when running the commands. WDYT?

Reply:
Or better yet, perhaps we should log whether we've timed out and/or we've hit some unknown error. Let me push a new commit to reflect this